pkg/devspace/build/build.go

```diff
@@ -98,16 +99,28 @@ func (c *controller) Build(ctx devspacecontext.Context, images []string, options
 
 			dockerClient, err := dockerclient.NewClient(ctx.Context(), ctx.Log())
 			if err == nil {
-				if imageConf.Kaniko != nil && imageConf.Kaniko.Namespace != "" && ctx.KubeClient().Namespace() != imageConf.Kaniko.Namespace {
-					err = pullsecrets.NewClient().EnsurePullSecret(ctx, dockerClient, imageConf.Kaniko.Namespace, registryURL)
+				if imageConf.Kaniko != nil {
+					pullSecretNamespace := ctx.KubeClient().Namespace()
+					if imageConf.Kaniko.Namespace != "" {
+						pullSecretNamespace = imageConf.Kaniko.Namespace
+					}
+
+					err = pullsecrets.NewClient().EnsurePullSecret(ctx, dockerClient, pullSecretNamespace, registryURL)
 					if err != nil {
 						ctx.Log().Errorf("error ensuring pull secret for registry %s: %v", registryURL, err)
 					}
 				}
 
-				err = pullsecrets.NewClient().EnsurePullSecret(ctx, dockerClient, ctx.KubeClient().Namespace(), registryURL)
-				if err != nil {
-					ctx.Log().Errorf("error ensuring pull secret for registry %s: %v", registryURL, err)
+				if imageConf.BuildKit != nil && imageConf.BuildKit.InCluster != nil {
```

Comment:
I don't think this is necessary as buildkit is using the local credentials automatically